Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup/Defs.lean

```diff
@@ -218,6 +219,27 @@ theorem coeToGL_det (g : SpecialLinearGroup n R) :
 @[simp]
 lemma coe_GL_coe_matrix (g : SpecialLinearGroup n R) : ((toGL g) : Matrix n n R) = g := rfl
 
+variable (S) in
+/-- `mapGL` is the map from the special linear group over `R` to the general linear group over
+`S`, where `S` is an `R`-algebra. -/
+def mapGL : Matrix.SpecialLinearGroup n R →* Matrix.GeneralLinearGroup n S :=
+  toGL.comp (map (algebraMap R S))
+
+@[simp]
+lemma mapGL_intCast_inj [CharZero R] (g g' : SpecialLinearGroup n ℤ) :
+    mapGL R g = mapGL R g' ↔ g = g' := by
+  simp [mapGL]
+
+lemma mapGL_intCast_injective [CharZero R] :
```

Comment:
Can you generalize this using `NoZeroSmulDivisors`?

Reply:
That's a good idea, thanks! I generalized it to assume `FaithfulSMul`, since that is what is assumed in the lemma `algebraMap_injective`.